lib/atlas/runner.rb

```diff
@@ -93,12 +93,10 @@ def query(string)
     end
 
     def set_graph(graph)
-      return graph if graph.is_a?(Turbine::Graph)
-
-      if @dataset.is_a?(Dataset::DerivedDataset)
-        @dataset.graph
+      if dataset.derived?
+        dataset.graph
```

Comment:
I think `graph` should still be preferred - also for derived datasets. If you explicitly specify a graph, then that should be used - at least it should not be silently ignored, i.e. take it or raise an exception.